scripts/extend_index_yaml_test.py

```diff
@@ -0,0 +1,309 @@
+                    f.write(index_yaml)
+                with open(
+                    MOCK_WEB_INF_INDEX_YAML_PATH, 'w', encoding='utf-8') as f:
+                    f.write(web_inf_index_yaml)
+                extend_index_yaml.main()
+                with open(MOCK_INDEX_YAML_PATH, 'r', encoding='utf-8') as f:
+                    actual_index_yaml = f.read()
+                self.assertEqual(actual_index_yaml, index_yaml)
+
+    def test_extend_index_yaml_with_same_kind(self):
+        index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: task
+  - name: status
+  - name: next_scheduled_check_time
+"""
+        web_inf_index_yaml = """indexes:
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
+"""
+        expected_index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: task
+  - name: status
+  - name: next_scheduled_check_time
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
+"""
+        with self.swap(
+            extend_index_yaml, 'INDEX_YAML_PATH',
+            MOCK_INDEX_YAML_PATH):
+            with self.swap(
+                extend_index_yaml, 'WEB_INF_INDEX_YAML_PATH',
+                MOCK_WEB_INF_INDEX_YAML_PATH):
+                with open(MOCK_INDEX_YAML_PATH, 'w', encoding='utf-8') as f:
+                    f.write(index_yaml)
+                with open(
+                    MOCK_WEB_INF_INDEX_YAML_PATH, 'w', encoding='utf-8') as f:
+                    f.write(web_inf_index_yaml)
+                extend_index_yaml.main()
+                with open(MOCK_INDEX_YAML_PATH, 'r', encoding='utf-8') as f:
+                    actual_index_yaml = f.read()
+                self.assertEqual(actual_index_yaml, expected_index_yaml)
+
+    def test_extend_index_yaml_with_same_kind_in_web_inf(self):
+        index_yaml = """indexes:
+
+- kind: AppFeedbackReportModel
+  properties:
+  - name: created_on
+  - name: scrubbed_by
+
+- kind: BlogPostRightsModel
+  properties:
+  - name: blog_post_is_published
+  - name: editor_ids
+  - name: last_updated
+    direction: desc
+"""
+        web_inf_index_yaml = """indexes:
+
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: task
+  - name: status
+  - name: next_scheduled_check_time
+
+- kind: ClassifierTrainingJobModel
+  properties:
+  - name: status
+  - name: next_scheduled_check_time
```

Comment:
Maybe one other thing that would be nice to check, is the same kind and the same properties but with properties in different order.

Reply:
Done.